chain/ethereum/src/network_indexer/subgraph.rs

```diff
@@ -88,8 +88,12 @@ fn create_subgraph(
         spec_version: String::from("0.0.1"),
         description: None,
         repository: None,
-        schema: Schema::parse(include_str!("./ethereum.graphql"), subgraph_id.clone())
-            .expect("valid Ethereum network subgraph schema"),
+        schema: Schema::parse(
+            include_str!("./ethereum.graphql"),
+            subgraph_id.clone(),
+            IdType::Bytes,
```

Comment:
Before I review the entire PR: Is this really a schema-wide setting and not a per-entity setting?

Reply:
It is, but only as far as expressing what should be `IdType::Bytes` is concerned; the `Layout` does not assume that it's schema-wide. If we want this to be per entity, we need to figure out a way how users express which entities use `IdType::Bytes`; I can think of two ways: either add a directive so that users say `type Foo { id: ID! @OnlyBytes }` or that we allow users to declare `id` as `Bytes`; either won't be much work, but I am not sure if using anything but type `ID` has ripple effects in other places - we would still represent `id` in an `Entity` as a Rust `String`; anything else would be a huge change all over the code base.

Reply:
Agreed. `id: ID! @bytes` or `id: Bytes!` could both work. Not sure right now, which one would be better.